virttest/utils_misc.py

```diff
@@ -608,6 +608,40 @@ def wait_for(func, timeout, first=0.0, step=1.0, text=None):
     return None
 
 
+def wait_for_ignore_errors(func, timeout, first=0.0, step=1.0, text=None):
```

Comment:
I assume you can merge those logic with previous wait_for function.
For example, define additional parameter ignore_exception = False
And in the except, you can choose raise or don't raise exception according to ignore_exception

Reply:
@chunfuwen That was my initial proposal, but that was rejected by @sathnaga So I proposed to have a second function instead, s. https://github.com/avocado-framework/avocado-vt/pull/2565#pullrequestreview-415236273

Reply:
ok

Reply:
Yeah, I agree. Adding a parameter to wait_for() seems bringing less redundancy. And what's even better about it is that future users could actually notice the 'ignore_errors' when they try to call the wait_for() func. Otherwise, there's good chance we will miss this "wait_for_ignore_errors" and continue to build a helper func with try-except by ourselves and use the old wait_for().

Reply:
@sathnaga Maybe you could reconsider allowing for the extra parameter?

Reply:
@smitterl make sense, am worried about if any regression this chance can introduce, as long as we do not break the normal flow, we are good with, pls go ahead with your previous approach, give it a test for few random tests using it....tnx!

Reply:
OK @sathnaga. I understand your worry about regression. I executed the tests

- with new parameter for verification https://github.com/autotest/tp-libvirt/pull/2795#issuecomment-632685785
- without new parameter for regression https://github.com/autotest/tp-libvirt/blob/6f2bbce4374c9e6c8b7cb7e3afe5af930a0ed943/libvirt/tests/src/virtual_device/watchdog.py
```bash
# avocado run --vt-type libvirt --vt-machine-type s390-virtio --vt-connect-uri qemu:///system watchdog --vt-no-filter watchdog..action_inject_nmi --vt-extra-params verify_host_dmesg=no
JOB ID     : e54547d6d302fcdd9c7bd018e05d744d24f8adb6
JOB LOG    : /root/avocado/job-results/job-2020-05-22T09.16-e54547d/job.log
 (1/6) type_specific.io-github-autotest-libvirt.watchdog.action_shutdown.model_diag288: PASS (51.14 s)
 (2/6) type_specific.io-github-autotest-libvirt.watchdog.action_dump.normal_domain_name.model_diag288: PASS (55.79 s)
 (3/6) type_specific.io-github-autotest-libvirt.watchdog.action_none.model_diag288: PASS (203.54 s)
 (4/6) type_specific.io-github-autotest-libvirt.watchdog.action_poweroff.model_diag288: PASS (53.20 s)
 (5/6) type_specific.io-github-autotest-libvirt.watchdog.action_pause.model_diag288: PASS (53.70 s)
 (6/6) type_specific.io-github-autotest-libvirt.watchdog.action_reset.model_diag288: PASS (70.91 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 490.53 s
```
https://github.com/autotest/tp-libvirt/blob/6f2bbce4374c9e6c8b7cb7e3afe5af930a0ed943/libvirt/tests/src/conf_file/qemu_conf/auto_dump.py
```bash
# avocado run --vt-type libvirt --vt-machine-type s390-virtio --vt-connect-uri qemu:///system conf_file.qemu_conf.auto_dump --vt-extra-params verify_host_dmesg=no
JOB ID     : 409a1db4cf90c02e981cf113a224f754b5e9a93c
JOB LOG    : /root/avocado/job-results/job-2020-05-22T09.40-409a1db/job.log
 (1/3) type_specific.io-github-autotest-libvirt.conf_file.qemu_conf.auto_dump.s390_model.default_bypass_cache: PASS (18.32 s)
 (2/3) type_specific.io-github-autotest-libvirt.conf_file.qemu_conf.auto_dump.s390_model.enable_bypass_cache: PASS (24.32 s)
 (3/3) type_specific.io-github-autotest-libvirt.conf_file.qemu_conf.auto_dump.s390_model.disable_bypass_cache: PASS (50.40 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 94.78 s
```

Reply:
Cool, Tnx!